decoder/gym.go

```diff
@@ -156,11 +158,32 @@ func calculatePowerUpPoints(fortData *pogo.PokemonFortProto) (null.Int, null.Int
 }
 
 func (gym *Gym) updateGymFromFort(fortData *pogo.PokemonFortProto, cellId uint64) *Gym {
+	type pokemonDisplay struct {
+		Form          int  `json:"form"`
+		Costume       int  `json:"costume"`
+		Gender        int  `json:"gender"`
+		Shiny         bool `json:"shiny"`
+		TempEvolution int  `json:"temp_evolution"`
+		Alignment     int  `json:"alignment"`
+		Badge         int  `json:"badge"`
+		LocationCard  int  `json:"location_card"`
+	}
 	gym.Id = fortData.FortId
 	gym.Lat = fortData.Latitude  //fmt.Sprintf("%f", fortData.Latitude)
 	gym.Lon = fortData.Longitude //fmt.Sprintf("%f", fortData.Longitude)
 	gym.Enabled = null.IntFrom(util.BoolToInt[int64](fortData.Enabled))
 	gym.GuardingPokemonId = null.IntFrom(int64(fortData.GuardPokemonId))
+	display, _ := json.Marshal(pokemonDisplay{
+		Form:          int(fortData.GuardPokemonDisplay.Form),
+		Costume:       int(fortData.GuardPokemonDisplay.Costume),
+		Gender:        int(fortData.GuardPokemonDisplay.Gender),
+		Shiny:         fortData.GuardPokemonDisplay.Shiny,
+		TempEvolution: int(fortData.GuardPokemonDisplay.CurrentTempEvolution),
+		Alignment:     int(fortData.GuardPokemonDisplay.Alignment),
+		Badge:         int(fortData.GuardPokemonDisplay.PokemonBadge),
+		LocationCard:  int(fortData.GuardPokemonDisplay.LocationCard.LocationCard),
```

Comment:
This causes `SIGSEGV: segmentation violation`... as it tries to access `LocationCard` which is not set

Reply:
Fixed ty.

Reply:
Is the fallback to 0 on purpose? What's the value though if available?

Reply:
I wonder if we also need to nil check the display in general, empty gyms? Can't test right now

Reply:
Hmm good point. `0` means unset so it is on purpose.